src/vs/workbench/contrib/notebook/browser/view/renderers/webviewPreloads.ts

```diff
@@ -984,7 +1068,8 @@ async function webviewPreloads(ctx: PreloadContext) {
 								cellId: preview.id,
 								container: preview,
 								isShadow: true,
-								originalRange: shadowSelection.getRangeAt(0)
+								originalRange: shadowSelection.getRangeAt(0),
+								searchPreviewInfo: extractSelectionLine(shadowSelection),
```

Comment:
`extractSelectionLine` can be costly, I wonder if we want to request this info conditionally (when requested from the search widget). Also we might want to handle the case where we have multiple search results on the same line, which we should optimize the DOM read operations to 1.

Reply:
We can probably handle this later, not necessary for this PR.

Reply:
oh actually it seems to run `extractSelectionLine` all the time, can we make sure that it's also behind the experimental flag?